src/bridge/mod.rs

```diff
@@ -52,6 +52,20 @@ pub struct Bridge<IS: AsyncRead + AsyncWrite + 'static> {
     joining_map: BTreeMap<String, String>,
 }
 
+fn handle_privmsg_action(text: &str) -> (String, String) {
```

Comment:
I find this harder to read than necessary. How about this instead:

```rust
let emote_content = text.strip_prefix(..).and_then(|noprefix| noprefix.strip_suffix(..));
match emote_content {
    Some(value) => ("m.emote", value.into()),
    None => ("m.text", text),
}
```

(and also make `text` a `String` so we don't needlessly copy it)

Reply:
Done, now using .and_then() to simplify the code.

Reply:
And now switched text to be a String. :-)